### PR TITLE
[SPARK-9449][SQL] Include MetastoreRelation's inputFiles

### PR DESCRIPTION
**Pull request by marmbrus**

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala [hunk 1]
```diff
@@ -516,6 +528,8 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
    */
   def paths: Array[String]
 
+  def inputFiles: Array[String] = cachedLeafStatuses().map(_.getPath.toString).toArray
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala]: override

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala [hunk 2]
```diff
@@ -888,6 +888,18 @@ private[hive] case class MetastoreRelation
   /** An attribute map for determining the ordinal for non-partition columns. */
   val columnOrdinals = AttributeMap(attributes.zipWithIndex)
 
+  def inputFiles: Array[String] = {
```
Comment (rxin) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: override

**Comment by marmbrus** [comment]
test this please

**Comment by marmbrus** [comment]
test this please

**Comment by rxin** [comment]
LGTM

**Comment by marmbrus** [comment]
Merged to master and 1.5

**Pull request closed without merging**